### PR TITLE
Iterate on optimization

### PR DESCRIPTION
**Pull request by tpoterba**
Thoughts?

### CODE REVIEW

**Comment by cseed** [comment]
1. I don't think this should be a user option.  Too fine grained.

2. init has an explosion of options.  I think we should have a centralized location for all the options and parameters.  The feature flag stuff is a step in that direction, but needs to be more pervasive.  Keep this all private/experimental for now.

I guess just 3 is going to get nearly all the benefit, but we'll need to track performance to know of sure.

**Comment by tpoterba** [comment]
I'll prefix the thing with an underscore.

I mostly want to expose this as an option so Chris can turn it back to 1 on the combiner

File: hail/python/hail/context.py [hunk 1]
```diff
@@ -27,12 +27,13 @@ class HailContext(object):
...
                       idempotent=bool,
                       global_seed=nullable(int),
+                      optimizer_iterations=nullable(int),
                       _backend=nullable(Backend))
     def __init__(self, sc=None, app_name="Hail", master=None, local='local[*]',
                  log=None, quiet=False, append=False,
                  min_block_size=1, branching_factor=50, tmp_dir=None,
                  default_reference="GRCh37", idempotent=False,
-                 global_seed=6348563392232659379, _backend=None):
+                 global_seed=6348563392232659379, optimizer_iterations=None, _backend=None):
```
Comment (cseed) [inline hail/python/hail/context.py]: Underscore here, too?

Comment (tpoterba) [reply hail/python/hail/context.py]: this isn't user-exposed

**Pull request merged**